test/test_extractor.py

```diff
@@ -53,74 +43,67 @@ def teardown_class(cls):
     async def extract_files(self, filenames):
         """Make sure all test files are present"""
         for filename in filenames:
-            assert os.path.exists(
-                os.path.join(self.tempdir, filename)
+            assert path.exists(
+                path.join(self.tempdir, filename)
             ), f"test file {filename} was not found"
         # Make sure extraction reports success
         for filename in filenames:
             async with self.extractor as ectx:
-                yield await ectx.aio_extract(os.path.join(self.tempdir, filename))
-
+                yield await ectx.aio_extract(path.join(self.tempdir, filename))
 
-class TestExtractor(TestExtractorBase):
-    """Test methods for the extractor functionality"""
+    @pytest.fixture
+    def extension_list(self) -> List[str]:
+        return []
 
-    def test_can_extract(self):
-        """Test that the can_extract function knows what it can do"""
-        assert self.extractor.can_extract(".tar.bz2")
-        assert self.extractor.can_extract(".zip")
-        assert self.extractor.can_extract(".deb")
+    @pytest.mark.asyncio
+    async def test_bad_files(self, extension_list: List[str]):
+        """Test handling of invalid files. No exceptions should be raised."""
+        for extension in extension_list:
+            filename = path.join(self.tempdir, f"empty-file{extension}")
+            open(filename, "w").close()
+            async for _ in self.extract_files([filename]):
```

Comment:
don't we need to explicitly assert if there's any exceptions?
I know this will work but IMO explicitly checking for exceptions will be better 

Related: https://github.com/intel/cve-bin-tool/issues/1331#issuecomment-895803640

Reply:
@BreadGenie, do you mean doing this?
```py
        for extension in extension_list:
            filename = path.join(self.tempdir, f"empty-file{extension}")
            open(filename, "w").close()
            try:
                async for _ in self.extract_files([filename]):
                    pass
            except subprocess.CalledProcessError:
                pytest.fail("Failed to extract file")
```